metrics.go

```diff
@@ -31,6 +31,9 @@ type namedMetric struct {
 type metric interface {
 	marshalTo(prefix string, w io.Writer)
 	metricType() string
+	// display: Whether this metric will be printed. it's based on whether sample is recorded.
```

Comment:
I don't like adding this interface:
1. It is needed only for metadat abut its name doesn't represent this
2. Its implementation should contain same logic as `marshalTo`, but it doesn't guarantee that. In future, this logic could diverse and introduce new bugs

The better way would be to keep the logic of whether metric must be renderred in one place, so it won't diverge in future.